internal/vault/vault.go

```diff
@@ -220,7 +221,7 @@ func (v *Vault) setToken(ctx context.Context, client Client) error {
 		return nil
 	}
 
-	return fmt.Errorf("error initializing Vault client: tokenSecretRef, appRoleSecretRef, or Kubernetes auth role not set")
+	return cmerrors.NewInvalidData("error initializing Vault client: tokenSecretRef, appRoleSecretRef, or Kubernetes auth role not set")
```

Comment:
Invalid data? I think "invalid config" is more appropriate here. Or is there an agreed cm-standard on this?

Reply:
Makes sense. Thanks for the explanation.

Reply:
@erikgb let's try to post a few additional Vault paths here that could also benefit from this `cmerrors.NewInvalidData` error.